### DISCUSSION TITLE
Compound `with` expression

### DISCUSSION THREAD

**Discussion by kyoyama-kazusa** [dotnet/csharplang#3965] [Language Ideas]
> I don't know whether anybody shares this idea, because I don't see anything about this idea. If duplicate, I'll be sorry for that.

Consider code:

```csharp
g = g with { Prop1 = DayOfWeek.Friday, Prop2 = 13 };
```

Simplification:

```csharp
g with= { Prop1 = DayOfWeek.Friday, Prop2 = 13 };
```

Complex pattern matching:

```csharp
if ((g with= { Prop1 = DayOfWeek.Friday, Prop2 = 13 }) is { ExampleProp: 20 })
{
    // Code playing with 'g'...
}
```

**Comment by huoyaoyuan** [comment]
I don't think putting compound assignment and condition in a single statement is necessary.
